src/listener/can_open_listener.py

```diff
@@ -101,14 +101,15 @@ def _read_complex_variable(self, sdo_client, sdo_index, node_id):
                 index_and_subindex = f'{sdo_index}sub{subindex}'
                 sdo_value = sdo_client.upload(sdo_index, subindex)
                 # Checks for every subindex if value changed
-                if(self._sdo_value_changed(index_and_subindex, sdo_value)):
+                if(self._sdo_value_changed(index_and_subindex, node_id,
+                                           sdo_value)):
                     self.inform_interpreter(sdo_value,
                          sdo_client[sdo_index][subindex].od.name, node_id,
                          hex(sdo_index), hex(subindex))
 
     def _read_simple_variable(self, sdo_client, sdo_index, node_id):
         sdo_value = sdo_client.upload(sdo_index, 0)
-        if(self._sdo_value_changed(sdo_index, sdo_value)):
+        if(self._sdo_value_changed(sdo_index, node_id, sdo_value)):
```

Comment:
moet er niet een subindex ook kunnen? Of is dat wat je bedoelt met "simple_variable"?

Reply:
simple variable heeft geen subindex. Omdat het geen array of record is. het is letterlijk een variable zoiets gezien wordt in de canopen documentatie